pyprep/find_noisy_channels.py

```diff
@@ -438,48 +439,104 @@ def find_bad_by_ransac(
                 " quality tests."
             )
 
-        # Make the ransac predictions
-        ransac_eeg = self.run_ransac(
-            chn_pos=chn_pos,
-            chn_pos_good=chn_pos_good,
-            good_chn_labs=good_chn_labs,
-            n_pred_chns=n_pred_chns,
-            data=self.EEGData,
-            n_samples=n_samples,
-        )
-
-        # Correlate ransac prediction and eeg data
-        correlation_frames = corr_window_secs * self.sample_rate
-        correlation_window = np.arange(correlation_frames)
-        n = correlation_window.shape[0]
-        correlation_offsets = np.arange(
-            0, (self.signal_len - correlation_frames), correlation_frames
-        )
-        w_correlation = correlation_offsets.shape[0]
+        try:
+            if channel_wise:
+                print("Forcing channel-wise ransac.")
+                raise MemoryError("Forcing channel-wise ransac.")
+            # Make the ransac predictions
+            ransac_eeg = self.run_ransac(
+                chn_pos=chn_pos,
+                chn_pos_good=chn_pos_good,
+                good_chn_labs=good_chn_labs,
+                n_pred_chns=n_pred_chns,
+                data=self.EEGData,
+                n_samples=n_samples,
+            )
 
-        # For the actual data
-        data_window = self.EEGData[: self.n_chans_new, : n * w_correlation]
-        data_window = data_window.reshape(self.n_chans_new, n, w_correlation)
+            # Correlate ransac prediction and eeg data
+            correlation_frames = corr_window_secs * self.sample_rate
+            correlation_window = np.arange(correlation_frames)
+            n = correlation_window.shape[0]
+            correlation_offsets = np.arange(
+                0, (self.signal_len - correlation_frames), correlation_frames
+            )
+            w_correlation = correlation_offsets.shape[0]
+
+            # For the actual data
+            data_window = self.EEGData[: self.n_chans_new, : n * w_correlation]
+            data_window = data_window.reshape(self.n_chans_new, n, w_correlation)
+
+            # For the ransac predicted eeg
+            pred_window = ransac_eeg[: self.n_chans_new, : n * w_correlation]
+            pred_window = pred_window.reshape(self.n_chans_new, n, w_correlation)
+
+            # Preallocate
+            channel_correlations = np.ones((w_correlation, self.n_chans_new))
```

Comment:
In this `Try` part, there are many lines of code that are then repeated in the `Except` part. Code duplication is almost never a good idea --> if you need a piece of code several times, make a short (private) function out of it, and re-use it.

Reply:
I will see how can generalize it, indeed is almost the same except for some crucial parameters.